Changes for rev1.0

asadate opened the pull request:
- Changed currentConsumption property into an array of objects since a component can have multiple supplies.
- Added into componentID.json a property to capture the digital datasheet specification revision referenced to generate an e-datasheet.
-  Initial assumption was that only supply pins required protections like OVP, UVLO. This is not the case. Expanded the definition of a powerSupply in ComponentProtectionThresholds.json file to include any pin (not just supply pins) that is monitored by protection mechanisms. Example, the CC lines in the USB Type-c PD controller may have over voltage protection.

asadate committed:
Update componentID.json to add digital specifications revisions 

Added the "eDatasheetSpecRevision" property to capture the digital datasheet specifications used to create the component digital datasheet. This is in anticipation of future revisions.

asadate committed:
Update componentProtectionThresholds.json to expand the definition of "supplyName"

First assumption was that only supply pins are protected. That's not always the case. Expanding the definition from supply pin to other pin gives the option to capture protection mechanisms any pin.

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Added current Consumption

asadate committed:
Change currentConsumption to array of objects

asadate committed:
Change currentConsumption to array of objects

asadate committed:
change currentConsumption into array of objects

asadate committed:
change currentConsumption into array of objects

asadate committed:
change currentConsumption into array of objects

asadate committed:
change currentConsumption into array of objects

asadate committed:
change currentConsumption into array of objects

asadate committed:
change currentConsumption into array of objects

asadate committed:
Update logic_gate.json

asadate committed:
Update dram.json

asadate committed:
Update eeprom.json

asadate committed:
Update flash_memory.json

asadate committed:
Update rom.json

asadate committed:
Update battery_charger.json

asadate committed:
Update displaybacklight_driver.json

asadate committed:
Update linear_regulator.json

asadate committed:
Update load_switch.json

asadate committed:
Update pmic.json

asadate committed:
Update switching_regulator.json

asadate committed:
Update accelerometer.json

asadate committed:
Update gyroscope.json

asadate committed:
Update magnetic_sensor.json

asadate committed:
Update thermal_sensor.json

asadate committed:
Update sd_card.json

asadate committed:
Update ssd.json

asadate committed:
Update undefined_ic.json

rsnancollas commented on part-spec/common/componentID.json:
```diff
@@ -106,11 +106,16 @@
                 },
+                "eDataheetSpecRevision": {
+                    "description": "revision of the digital datasheet specifications used to create the digital datasheet",
+                    "example": "rev 1.0",
```
suggest that we make this an enum (where we add a string each time we release) - that way we won't get revisions that we can't match against anything.

so "enum": ["1.0"]

asadate replied:
Changed

rsnancollas commented on part-spec/ic_microcontroller/microcontroller.json:
```diff
@@ -53,9 +53,12 @@
                 "type": "string"
             },
-            "currentConsumption": {
+             "currentConsumption": {
```
why did you add a space?

asadate replied:
changed

rsnancollas reviewed (changes requested):
mostly looks good - one change suggested for the eDatasheetSpecRevision.

asadate committed:
Update componentID.json

Changed eDatasheetSpecRevision to enum

asadate committed:
Update microcontroller.json

ajayitem commented on part-spec/common/componentID.json:
```diff
@@ -106,11 +106,16 @@
                     "type": "string"
                 },
+                "eDataheetSpecRevision": {
+                    "description": "revision of the digital datasheet specifications used to create the digital datasheet",
```
There's a spelling error here "eDataheetSpecRevision". It should be "eDatasheetSpecRevision"

asadate replied:
Good Catch. Changed

asadate committed:
Update componentID.json

asadate commented:
Changed currentConsumption property into an array of objects since a component can have multiple supplies.
Added into componentID.json a property to capture the digital datasheet specification revision referenced to generate an e-datasheet.
Initial assumption was that only supply pins required protections like OVP, UVLO. This is not the case. Expanded the definition of a powerSupply in ComponentProtectionThresholds.json file to include any pin (not just supply pins) that is monitored by protection mechanisms. Example, the CC lines in the USB Type-c PD controller may have over voltage protection.
